service-impl/audit-service-impl/stroom-audit-service/src/main/java/uk.gov.gchq.palisade.audit.service.impl/StroomSimpleAuditService.java

```diff
@@ -0,0 +1,249 @@
+        });
+        //handler for ReadRequestExceptionAuditRequest
+        DISPATCH.put(ReadRequestExceptionAuditRequest.class, (o) -> {
+            requireNonNull(o, "ReadRequestExceptionAuditRequest");
+            ReadRequestExceptionAuditRequest readRequestExceptionAuditRequest = (ReadRequestExceptionAuditRequest) o;
+            final String msg = " 'readRequestException' "
+                    + readRequestExceptionAuditRequest.getId() + " "
+                    + readRequestExceptionAuditRequest.getRequestId() + " "
+                    + readRequestExceptionAuditRequest.getOriginalRequestId() + " "
+                    + readRequestExceptionAuditRequest.getResource().toString() + " "
+                    + readRequestExceptionAuditRequest.getException().toString();
+            logToStroom(ReadRequestExceptionAuditRequest.class, msg, readRequestExceptionAuditRequest.getRequestId(), null);
+        });
+        //handler for ReadRequestReceivedAuditRequest
+        DISPATCH.put(ReadRequestReceivedAuditRequest.class, (o) -> {
+            requireNonNull(o, "ReadRequestReceivedAuditRequest");
+            ReadRequestReceivedAuditRequest readRequestReceivedAuditRequest = (ReadRequestReceivedAuditRequest) o;
+            final String msg = " 'readRequestReceived' "
+                    + readRequestReceivedAuditRequest.getId() + " "
+                    + readRequestReceivedAuditRequest.getRequestId() + " "
+                    + readRequestReceivedAuditRequest.getOriginalRequestId() + " "
+                    + readRequestReceivedAuditRequest.getResource().toString() + " ";
+            logToStroom(ReadRequestReceivedAuditRequest.class, msg, readRequestReceivedAuditRequest.getRequestId(), null);
+        });
+        //handler for ReadResponseAuditRequest
+        DISPATCH.put(ReadResponseAuditRequest.class, (o) -> {
+            requireNonNull(o, "ReadResponseAuditRequest");
+            ReadResponseAuditRequest readResponseAuditRequest = (ReadResponseAuditRequest) o;
+            final String msg = " 'readResponseAuditRequest' "
+                    + readResponseAuditRequest.getId() + " "
+                    + readResponseAuditRequest.getRequestId() + " "
+                    + readResponseAuditRequest.getOriginalRequestId() + " "
+                    + readResponseAuditRequest.getResource().toString() + " ";
+            logToStroom(ReadResponseAuditRequest.class, msg, readResponseAuditRequest.getRequestId(), null);
+        });
+    }
+
+    public StroomSimpleAuditService() {
+        //Create the EVENT_LOGGING_SYSTEM that is logging the authenticat event
+        EVENT_LOGGING_SYSTEM.setName("Palisade Audit Service");
+        EVENT_LOGGING_SYSTEM.setEnvironment("Test");
+        String hostName = "";
+        String hostAddress = "";
+
+
+    }
+
+    static String auditLogContext(final AuditRequestWithContext auditRequestWithContext) {
```

Comment:
Should this be package visible? public or private seems more appropriate. If making it public, please null check the method parameter.